src/TorchSharp/TorchVision/Ops/Misc.cs

```diff
@@ -0,0 +1,145 @@
+
+// A number of implementation details in this file have been translated from the Python version of torchvision,
+// largely located in the files found in this folder:
+//
+// https://github.com/pytorch/vision/blob/a4f53308b2d0f1aa9191686e326f45c26053f686/torchvision/ops/misc.py
+//
+// The origin has the following copyright notice and license:
+//
+// https://github.com/pytorch/vision/blob/main/LICENSE
+//
+
+using System;
+using System.Collections.Generic;
+using static TorchSharp.torch;
+
+#nullable enable
+namespace TorchSharp
+{
+    public static partial class torchvision
+    {
+        public static partial class ops
+        {
+            /// <summary>
+            /// Configurable block used for Convolution2d-Normalization-Activation blocks.
+            /// </summary>
+            /// <param name="in_channels">Number of channels in the input image</param>
+            /// <param name="out_channels">Number of channels produced by the Convolution-Normalization-Activation block</param>
+            /// <param name="kernel_size">Size of the convolving kernel.</param>
+            /// <param name="stride">Stride of the convolution.</param>
+            /// <param name="padding">Padding added to all four sides of the input. Default: null, in which case it will calculated as ``padding = (kernel_size - 1) // 2 * dilation``</param>
+            /// <param name="groups">Number of blocked connections from input channels to output channels.</param>
+            /// <param name="norm_layer">Norm layer that will be stacked on top of the convolution layer. If ``null``, ``torch.nn.BatchNorm2d`` is used.</param>
+            /// <param name="activation_layer">Activation function which will be stacked on top of the normalization layer (if not None), otherwise on top of the conv layer. If ``null``, ``torch.nn.ReLU`` is used.</param>
+            /// <param name="dilation">Spacing between kernel elements.</param>
+            /// <param name="inplace">Parameter for the activation layer, which can optionally do the operation in-place.</param>
+            /// <param name="bias">Whether to use bias in the convolution layer. By default, is false.</param>
+            public static nn.Module Conv2dNormActivation(
+                long in_channels,
+                long out_channels,
+                long kernel_size = 3,
+                long stride = 1,
+                long? padding = null,
+                long groups = 1,
+                Func<long, nn.Module>? norm_layer = null,
+                Func<bool, nn.Module>? activation_layer = null,
+                long dilation = 1,
+                bool inplace = true,
+                bool? bias = null)
+            {
+                if (padding == null) {
+                    padding = (kernel_size - 1) / 2 * dilation;
+                }
+
+                if (bias == null) {
```

Comment:
Given that 'null' doesn't seem to mean anything special, why can't the parameter just be a 'bool' with 'false' as the default?

Reply:
I changed the code not to fall back to `torch.nn.BatchNorm2d` when `norm_layer` is `null`.

The original signature of torchvision is as below. It accepts the optional `norm_layer` with a default callable. In C# the default value must be `null` as it must be compile time static, but I wanted to emulate the behavior.

https://github.com/pytorch/vision/blob/841b9a19a5ae6fb92b517438061b9846153cc0c6/torchvision/ops/misc.py#L144

```
    def __init__(
        self,
        in_channels: int,
        out_channels: int,
        kernel_size: Union[int, Tuple[int, int]] = 3,
        stride: Union[int, Tuple[int, int]] = 1,
        padding: Optional[Union[int, Tuple[int, int], str]] = None,
        groups: int = 1,
        norm_layer: Optional[Callable[..., torch.nn.Module]] = torch.nn.BatchNorm2d,
        activation_layer: Optional[Callable[..., torch.nn.Module]] = torch.nn.ReLU,
        dilation: Union[int, Tuple[int, int]] = 1,
        inplace: Optional[bool] = True,
        bias: Optional[bool] = None,
    ) -> None:
```